### PR TITLE
feat: ES 7.x client support

### PR DESCRIPTION
**Pull request by jywadhwani**
This involves ES client side changes to support ES 7.9.3 (we are currently at version 5.6.8). A separate module is created which makes use of ES 7.9.3 client.

## Checklist

- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [X] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by jplaisted** [comment]
Were there any code changes required? Or were the only changes required build files?

**Comment by jywadhwani** [comment]
> Were there any code changes required? Or were the only changes required build files?

There were code changes as well.

**Comment by jplaisted** [comment]
Can you describe them in commit / PR description? would make this review easier.

**Comment by jywadhwani** [comment]
> Can you describe them in commit / PR description? would make this review easier.

Added. Let me know if there is a better way to describe the code changes.

**Comment by jplaisted** [comment]
I think this is fine for now.

Thinking about it, a "better" way, if we do something similar in the future, would be to have a PR with two commits. The first commit just verbatim copies the code. The second commit modifies it. Then I can view the diff between the two commits. Not worth messing around with for now, but we can consider that in the future.

File: dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+public class ESSearchDAOTest {
+
+  private ESSearchDAO<EntityDocument> _searchDAO;
+  private ESAutoCompleteQueryForHighCardinalityFields _esAutoCompleteQuery;
+  private TestSearchConfig _testSearchConfig;
+
+  private static String loadJsonFromResource(String resourceName) throws IOException {
+    String jsonStr = IOUtils.toString(ClassLoader.getSystemResourceAsStream(resourceName), StandardCharsets.UTF_8);
+    return jsonStr.replaceAll("\\s+", "");
```
Comment (jplaisted) [inline dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java:49]: Why was this changed? Does Elasticsearch no longer accept whitespace in its json?

Comment (jywadhwani) [reply dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java:49]: The JSON returned by `searchRequest.source()` was single-line with no whitespace. On the other hand, JSON read by the method `loadJsonFromResource` earlier was in multi-line form. Hence needed the conversion for asserts to work.

**Comment by jywadhwani** [comment]
> I think this is fine for now.
> 
> Thinking about it, a "better" way, if we do something similar in the future, would be to have a PR with two commits. The first commit just verbatim copies the code. The second commit modifies it. Then I can view the diff between the two commits. Not worth messing around with for now, but we can consider that in the future.

Agree, worth considering in future.

File: dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/browse/BrowseDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    assertEquals(ESBrowseDAO.getNextLevelPath(browsePaths, path5),
+        "/School/Characteristics/General/Embedding/network_standardized_school_embeddings_v3");
+
+    // Scenario 6: current path isn't normalized, which ideally should not be the case
+    String path6 = "/School/Characteristics/General/Embedding";
+    assertEquals(ESBrowseDAO.getNextLevelPath(browsePaths, path6),
+        "/School/Characteristics/General/Embedding/network_standardized_school_embeddings_v3");
+  }
+
+
```
Comment (jywadhwani) [inline dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/browse/BrowseDAOTest.java:71]: Need to enable test for this method, for some reason it got disabled. Working on it.

File: dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java (hunk 1 above)
Comment (jplaisted) [reply dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java:49]: Ah, ok, this isn't used for requests, but verifying responses, got it.

**Comment by jplaisted** [comment]
fyi, with github, you don't need to do amend / force pushes. in fact, it's probably bad. I can't see what you've changed in that last push, unlike with review board.

When you merge the PR you have the opportunity to squash all the commits in the PR, so have many small commits with comments like "address comments" are totally fine since you'll wipe them out later.

tl;dr review board = one commit ever, github = each change is a new commit (please), clean up at submit

**Comment by jywadhwani** [comment]
> fyi, with github, you don't need to do amend / force pushes. in fact, it's probably bad. I can't see what you've changed in that last push, unlike with review board.
> 
> When you merge the PR you have the opportunity to squash all the commits in the PR, so have many small commits with comments like "address comments" are totally fine since you'll wipe them out later.
> 
> tl;dr review board = one commit ever, github = each change is a new commit (please), clean up at submit
> 
> edit: for example, check out #12. Many commits in the PR, but only one final commit in the history when it was merged.

The last commit was related to `BrowseDAOTest`.
Regardless, I have separated changes to multiple commits to make the review easier. Hope this addresses your concerns.

**Review by jplaisted — APPROVED** [review]
LGTM, be sure to do "squash and merge", then clean up the description.
